### ISSUE TITLE
`Array.eval` does not set pre-initialized output (possibly applies to any DataSource)

### ISSUE THREAD

**Issue by jmilloy** [creare-com/podpac#197]
**Description**

When you provide an `Array` data source node with an pre-initialized output array, the data is not set, .

**Steps to Reproduce**

```
import numpy as np
from podpac.data import Array
from podpac.coordinates import Coordinates, clinspace

coords = Coordinates([clinspace(-25, 25, 100), clinspace(-25, 25, 100)], dims=['lat', 'lon'])
node = Array(source=np.linspace(0., 100., 10000).reshape(100, 100), native_coordinates=coords)
out = node.create_output_array(coords)
out2 = node.eval(coords, output=out)
out3 = node.eval(coords)
```

**Expected Behavior**

 * `out` should contain the evaluated data (in fact, `out`, `out2`, and `out3` should all be equal)
 * `out` and `out2` should be the *same* object.

**Observed Behavior**

 * `out` is all np.nan (`out2` and `out3` do contain the evaluated data)
 * `out` and `out2` are different objects.

**Additional Notes**

 * I would be surprised if this is limited to the `Array` node. It does not have an `eval` method, but just inherits `DataSource.eval`.
 * The second requirement, that `out` and `out2` be the same object is less important.

**Comment by jmilloy** [comment]
This primary issue (the output argument not being set) is due to datasource.py:337

```
335:        # set the order of dims to be the same as that of evaluated coordinates
336:        # this is required in case the user supplied an output object with a different dims order
337:        output = output.transpose(*coordinates.dims)
```

**Comment by jmilloy** [comment]
@mlshapiro I'm assigning you, but bounce this back to me if you need to.

See also https://github.com/creare-com/podpac/commit/57a1aee239a6ad566def6ac86b7549ba8dffdfad

**Comment by jmilloy** [comment]
Possibly what we need to do is transpose the `coordinates` to match the provided output (instead of transposing the output to match the provided coordinates). We'd need to keep remember the order so that we can transpose the output back before returning. Actually, now that I've thought that through, let me just try that.

**Comment by jmilloy** [comment]
See #198

**Comment by mlshapiro** [comment]
@jmilloy is this closed by your PR #198 ?

**Comment by jmilloy** [comment]
Yes
